packages/http/src/validator/utils/spec.ts

```diff
@@ -5,8 +5,22 @@ export function findOperationResponse(
   statusCode: number,
 ): IHttpOperationResponse | undefined {
   const sortedSpecs = responseSpecs
-    .filter(spec => new RegExp(`^${spec.code.replace(/X/g, '\\d')}$`).test(String(statusCode)))
-    .sort((s1, s2) => s1.code.split('X').length - s2.code.split('X').length);
+    .filter(
+      spec =>
+        new RegExp(`^${spec.code.replace(/X/g, '\\d')}$`).test(String(statusCode)) ||
+        spec.code.toLowerCase() === 'default',
+    )
+    .sort((s1, s2) => {
+      if (s1.code.toLowerCase() === 'default') {
+        return 1;
+      }
+
+      if (s2.code.toLowerCase() === 'default') {
+        return -1;
```

Comment:
Same question — shouldn't this be 1 since we want default to be a fallback choice?

Reply:
Same here